src/main/java/com/spring/test/dao/UserDaoImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.spring.test.dao;
+
+import com.spring.test.exception.DataProcessingException;
+import com.spring.test.model.User;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    @Autowired
+    private SessionFactory sessionFactory;
```

Comment:
What is better: setter auto wiring, constructor auto wiring, or field injection?

Reply:
constructor